pytorch_lightning/core/step_result.py

```diff
@@ -129,7 +129,14 @@ def log(
     ):
         # no metrics should be logged with graphs
         if not enable_graph and isinstance(value, torch.Tensor):
-            value = value.detach()
+            # TODO: Find a way to make the reduction only once, so we don't need to clone the
+            is_dist = torch.distributed.is_available() and torch.distributed.is_initialized()
+            if sync_dist and is_dist:
+                # make sure we don't make an in-place operation
+                # .detach().clone() and not .clone().detach() which will copy the entire graph.
+                value = value.detach().clone()
```

Comment:
What's preventing the clone() from being in the `if sync_dist` block directly below?

Reply:
Great catch ! Nothing really :)

Reply:
Gotcha! and is that what you meant by this? I'm not too familiar with the training loop after all these refactors haha
> Find a way to make the sync_dist only once.